go/tasks/pluginmachinery/flytek8s/pod_template_store.go

```diff
@@ -0,0 +1,84 @@
+package flytek8s
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var DefaultPodTemplateStore PodTemplateStore = NewPodTemplateStore()
+
+// PodTemplateStore maintains a thread-safe mapping of active PodTemplates with their associated
+// namespaces.
+type PodTemplateStore struct {
+	defaultNamespace      string
+	mutex                 sync.Mutex
+	namespacePodTemplates map[string]*v1.PodTemplate
```

Comment:
I think using `sync.Map` may offer better (lower) contentions...
https://pkg.go.dev/sync#Map

Reply:
Much better, didn't know about the thread-safe libs. Going to have to take a deeper look.